### PR TITLE
upgrade to latest storybook and support React 16

### PR DESCRIPTION
**Pull request by DominikGuzei**
This PR modernizes the setup by upgrading to the latest storybook version (and to webpack 2) available and adding support for React 16. It also cleans up the npm dependencies and pins important packages to specific versions.

screenshot 2017-10-11 um 15 15 20

### CODE REVIEW

**Comment by nikolaglumac** [comment]
@DominikGuzei how do you install dependencies? Via npm or using yarn?
I have used `npm install` and everything seems to be working but there were some unmet dependencies warnings:
screen shot 2017-10-11 at 17 50 32

File: storybook/webpack.config.js [hunk 1]
```diff
@@ -1,30 +1,33 @@
-const findCacheDir = require('find-cache-dir');
-
 module.exports = {
```
Comment (darko-mijic) [inline storybook/webpack.config.js]: You can use ES2015 module syntax here. Why not webpack 3?

File: storybook/webpack.config.js [hunk 2]
```diff
@@ -1,30 +1,33 @@
...
+            ],
+          }
+        }],
+      },
+      {
         test: /\.scss$/,
-        loaders: [
+        use: [
           'style-loader?sourceMap',
           'css-loader?sourceMap&modules&localIdentName=[name]_[local]&importLoaders=1',
```
Comment (darko-mijic) [inline storybook/webpack.config.js]: `options: {}` in new webpack syntax look nicer.

File: storybook/webpack.config.js (hunk 1 above)
Comment (DominikGuzei) [reply storybook/webpack.config.js]: No in the config it's normal node.js code. Latest storybook is using webpack 2 😉

File: storybook/webpack.config.js (hunk 2 above)
Comment (DominikGuzei) [reply storybook/webpack.config.js]: Good point, is updated! 👍

**Pull request merged**